uc-crux-llvm/src/UCCrux/LLVM/Run/Unsoundness.hs

```diff
@@ -39,39 +46,60 @@ data WithUnsoundness a = WithUnsoundness
 
 data Unsoundness = Unsoundness
   { unsoundOverridesUsed :: Set UnsoundOverrideName,
-    unsoundSkipOverridesUsed :: Set SkipOverrideName
+    unsoundSkipOverridesUsed :: Set SkipOverrideName,
+    unsoundSpecsUsed :: Set SpecUse
   }
   deriving (Eq, Ord, Show)
 
-ppUnsoundness :: Unsoundness -> Doc Void
-ppUnsoundness u =
+ppUnsoundness ::
+  -- | What kind of unsoundness do you care about? For instance, if this is
+  -- 'Soundness.Overapprox', this function will print only things that are
+  -- unsound for verification.
+  Soundness ->
+  Unsoundness ->
+  Doc Void
+ppUnsoundness s u =
   PP.nest 2 $
     PP.vcat $
-      ( PP.pretty
-          "The following unsound overrides (built-in functions) were used:" :
-        bullets
-          (map getUnsoundOverrideName (Set.toList (unsoundOverridesUsed u)))
-      )
-        ++ ( PP.pretty
-               "Execution of the following functions was skipped:" :
-             bullets
-               (map getSkipOverrideName (Set.toList (unsoundSkipOverridesUsed u)))
-           )
+      concat
+        [ PP.pretty
+            "The following unsound overrides (built-in functions) were used:" :
+              bullets
+                (map getUnsoundOverrideName (Set.toList (unsoundOverridesUsed u)))
+        , PP.pretty
+            "Execution of the following functions was skipped:" :
+              bullets
+                (map getSkipOverrideName (Set.toList (unsoundSkipOverridesUsed u)))
+        , PP.pretty
+            "The following unsound specifications were applied:" :
+              bullets specs
+        ]
   where
     bullets = map ((PP.pretty "-" PP.<+>) . PP.pretty)
```

Comment:
Aesthetics opinion: it would be nice if `bullets` printed something like "[None]" if the argument was empty instead of having the header immediately followed by the next header.

Reply:
Done, good suggestion!